src/test-helpers/helpers.ts

```diff
@@ -0,0 +1,87 @@
+import type { JsonMap } from '@pact-foundation/pact/src/common/jsonTypes'
+
+/**
+ * Converts an object with arbitrary value types to a `JsonMap` where all values are compatible with Pact's expectations.
+ * This is useful for ensuring compatibility with functions that expect all values in the map to be of specific types,
+ * such as in Pact state management.
+ *
+ * The function handles various data types as follows:
+ * - `null` and `undefined`: Converted to the string `"null"`.
+ * - `object`: Serialized using `JSON.stringify` unless it's a Date or Array.
+ * - `number` and `boolean`: Preserved as is, ensuring numeric and boolean types remain unaltered.
+ * - `Date`: Converted to an ISO string format (`toISOString()`).
+ * - `Array`: Preserved in its original form, as arrays should not be converted to strings.
+ *
+ * @param {Record<string, unknown>} obj - The object to convert, with string keys and values of any type.
+ * @returns {JsonMap} - A new object where all values are in a format compatible with Pact.
+ *
+ * @example
+ * const movie = { name: 'Inception', year: 2010, released: new Date(), tags: ['Sci-Fi', 'Thriller'], director: { firstName: 'Christopher', lastName: 'Nolan' } }
+ * toJsonMap(movie) // { name: 'Inception', year: 2010, released: '2024-09-01T12:00:00.000Z', tags: ['Sci-Fi', 'Thriller'], director: '{"firstName":"Christopher","lastName":"Nolan"}' }
```

Comment:
I think this is a mistake?

> { name: 'Inception', year: 2010, released: '2024-09-01T12:00:00.000Z', tags: ['Sci-Fi', 'Thriller'], director: '{"firstName":"Christopher","lastName":"Nolan"}' }

Note that `director` is a `String` but I think it should be an object?